elastic/kibana
# [Index management] Fix to enable adding new fields with new inference endpoint created from inference Flyout

## Summary
Fixes bug, enabling adding a new fields with inference endpoints created from inference flyout. 


**Background**

This [PR](https://github.com/elastic/kibana/pull/180330) introduced a package `@kbn/inference_integration_flyout`  to enable adding a new inference endpoint with Elasticsearch models - ELSER & E5, Third party model - HuggingFace, Cohere, OpenAI and uploading via ELAND instructions in a Flyout. 

This flyout component is used by Add field component, when it's a `semantic_text` field and would like to create a new inference endpoint id for the field. 

**Bug description**

After a new inference endpoint is created, Cannot add field with this inference endpoint . 

https://github.com/elastic/kibana/assets/55930906/ad62fdbb-4ef6-40af-9812-5d91b7e63c26


**Expected** 

Should be able to add a new field with this new inference endpoint 


**Testing instructions**

**Elasticsearch changes (only to test save mappings)**

Since ES changes for the semantic_text has been merged to main, this can be tested against running ES from source or from latest snapshot
1. Update local branch with latest Elasticsearch changes from main
2. Run the elasticsearch: `./gradlew :run -Drun.license_type=trial`


**Manual test in UI**

1. Set is [isSemanticTextEnabled](https://github.com/elastic/kibana/blob/e89b991d7473caba2a3a5b6204080f50100c67b9/x-pack/plugins/index_management/public/application/sections/home/index_list/details_page/details_page_mappings_content.tsx#L72) to true 
4. Add a new field with type - `Semantic_text` 
5. Click on drop down menu below `Select an inference endpoint:` 
6. Click `Add inference Endpoint` 
7.  create a new inference endpoint in Add inference endpoint flyout 
8. the drop down menu list, from Step 3, should have the new inference endpoint created on Step 5
9. Fill in the reference field and Field name
10.  click add field
11. A new field should be created using this inference endpoint
